syntax = "proto2";
import "nanopb.proto";

message DataMessage {
  required uint32 id = 1;
  required uint32 sample_number = 2;
  required uint32 unix_time = 3;
  required uint32 unix_time_nsecs = 4;
  required uint32 time_uncertainty = 5;
  required float Data_01 =6;
  optional float Data_02 =7;
  optional float Data_03 =8;
  optional float Data_04 =9;
  optional float Data_05 =10;
  optional float Data_06 =11;
  optional float Data_07 =12;
  optional float Data_08 =13;
  optional float Data_09 =14;
  optional float Data_10 =15;
  optional float Data_11 =16;
  optional float Data_12 =17;
  optional float Data_13 =18;
  optional float Data_14 =19;
  optional float Data_15 =20;
  optional float Data_16 =21;
}

message DescriptionMessage {
  required uint32 id = 1;
  required string Sensor_name = 2 [(nanopb).max_size = 40];
  enum  DESCRIPTION_TYPE{
  PHYSICAL_QUANTITY = 0;
  UINT = 1;
  UNCERTAINTY_TYPE = 2;
  RESOLUTION = 3;
  MIN_SCALE = 4;
  MAX_SCALE = 5;
  }
  required DESCRIPTION_TYPE Description_Type =3;
  optional string str_Data_01 =4 [(nanopb).max_size = 40];
  optional string str_Data_02 =5 [(nanopb).max_size = 40];
  optional string str_Data_03 =6 [(nanopb).max_size = 40];
  optional string str_Data_04 =7 [(nanopb).max_size = 40];
  optional string str_Data_05 =8 [(nanopb).max_size = 40];
  optional string str_Data_06 =9 [(nanopb).max_size = 40];
  optional string str_Data_07 =10 [(nanopb).max_size = 40];
  optional string str_Data_08 =11 [(nanopb).max_size = 40];
  optional string str_Data_09 =12 [(nanopb).max_size = 40];
  optional string str_Data_10 =13 [(nanopb).max_size = 40];
  optional string str_Data_11 =14 [(nanopb).max_size = 40];
  optional string str_Data_12 =15 [(nanopb).max_size = 40];
  optional string str_Data_13 =16 [(nanopb).max_size = 40];
  optional string str_Data_14 =17 [(nanopb).max_size = 40];
  optional string str_Data_15 =18 [(nanopb).max_size = 40];
  optional string str_Data_16 =19 [(nanopb).max_size = 40];
  optional float f_Data_01 =20;
  optional float f_Data_02 =21;
  optional float f_Data_03 =22;
  optional float f_Data_04 =23;
  optional float f_Data_05 =24;
  optional float f_Data_06 =25;
  optional float f_Data_07 =26;
  optional float f_Data_08 =27;
  optional float f_Data_09 =28;
  optional float f_Data_10 =29;
  optional float f_Data_11 =30;
  optional float f_Data_12 =31;
  optional float f_Data_13 =32;
  optional float f_Data_14 =33;
  optional float f_Data_15 =34;
  optional float f_Data_16 =35;
}
